tst/EnergyPlus/unit/WeatherManager.unit.cc

```diff
@@ -151,6 +151,37 @@ TEST_F(EnergyPlusFixture, SkyTempTest)
     EXPECT_NEAR(3.02, TomorrowSkyTemp(1, 1), .001);
 }
 
+TEST_F(EnergyPlusFixture, SkyEmissivityTest)
+{
+    // setup environment state
+    Environment.allocate(4);
+    Environment(1).SkyTempModel = WP_ClarkAllenModel;
+    Environment(2).SkyTempModel = WP_BruntModel;
+    Environment(3).SkyTempModel = WP_IdsoModel;
+    Environment(4).SkyTempModel = WP_BerdahlMartinModel;
+
+    // init local variables
+    Real64 OpagueSkyCover(0.0);
+    Real64 DryBulb(25.0);
+    Real64 DewPoint(16.7);
+    Real64 RelHum(0.6);
+
+    EXPECT_NEAR(0.832, CalcSkyEmissivity(Environment(1).SkyTempModel, OpagueSkyCover, DryBulb, DewPoint, RelHum), 0.001);
+    EXPECT_NEAR(0.862, CalcSkyEmissivity(Environment(2).SkyTempModel, OpagueSkyCover, DryBulb, DewPoint, RelHum), 0.001);
+    EXPECT_NEAR(0.867, CalcSkyEmissivity(Environment(3).SkyTempModel, OpagueSkyCover, DryBulb, DewPoint, RelHum), 0.001);
+    EXPECT_NEAR(0.862, CalcSkyEmissivity(Environment(4).SkyTempModel, OpagueSkyCover, DryBulb, DewPoint, RelHum), 0.001);
+
+    DryBulb = 5.0;
+    DewPoint = -2.13;
+    EXPECT_NEAR(0.781, CalcSkyEmissivity(Environment(1).SkyTempModel, OpagueSkyCover, DryBulb, DewPoint, RelHum), 0.001);
+    EXPECT_NEAR(0.746, CalcSkyEmissivity(Environment(2).SkyTempModel, OpagueSkyCover, DryBulb, DewPoint, RelHum), 0.001);
+    EXPECT_NEAR(0.760, CalcSkyEmissivity(Environment(3).SkyTempModel, OpagueSkyCover, DryBulb, DewPoint, RelHum), 0.001);
+    EXPECT_NEAR(0.747, CalcSkyEmissivity(Environment(4).SkyTempModel, OpagueSkyCover, DryBulb, DewPoint, RelHum), 0.001);
+
+    Environment.deallocate();
```

Comment:
Should not be necessary to deallocate here.  I know it can be seen as a safeguard, but extra deallocations have a tendency to mask deallocations that should be in the appropriate `clear_state` functions.